src/omotes_simulator_core/entities/network_controller.py

```diff
@@ -244,4 +277,8 @@ def _set_producers_based_on_priority(self, time: datetime.datetime) -> dict:
                     if source.priority == priority:
```

Comment:
This part of the loop you check two times could this be simplified? Instead of checkking all producers have an indexed list where you just put in the priority to get the associated assets? You are performing the check multiple times!

Reply:
There can be multiple sources with same priority, But I can just make a list of the source with the priority to check, that makes it alreayd better